### PR TITLE
feat: Add support for codemirror mode languages

### PR DESCRIPTION
**Pull request by KonradStanski**
Codemirror 5 had a number of languages supported via modes instead of explicit language support.
These are the languages here:
https://codemirror.net/5/mode/
Fixes: https://github.com/sourcebot-dev/sourcebot/issues/110

### CODE REVIEW

File: packages/web/src/hooks/useSyntaxHighlightingExtension.ts [hunk 1]
```diff
@@ -2,20 +2,68 @@
...
+import { erlang } from "@codemirror/legacy-modes/mode/erlang";
+import { fortran } from "@codemirror/legacy-modes/mode/fortran";
+import { groovy } from "@codemirror/legacy-modes/mode/groovy";
+import { haskell } from "@codemirror/legacy-modes/mode/haskell";
+import { jinja2 } from "@codemirror/legacy-modes/mode/jinja2";
+import { julia } from "@codemirror/legacy-modes/mode/julia";
+import { liveScript } from "@codemirror/legacy-modes/mode/livescript";
+import { lua } from "@codemirror/legacy-modes/mode/lua";
+import { nginx } from "@codemirror/legacy-modes/mode/nginx";
+// import { octave } from "@codemirror/legacy-modes/mode/octave";
```
Comment (KonradStanski) [inline packages/web/src/hooks/useSyntaxHighlightingExtension.ts:49]: These exist on the https://codemirror.net/5/mode/ page but are seemingly missing from the import.

File: packages/web/src/hooks/useSyntaxHighlightingExtension.ts [hunk 2]
```diff
@@ -2,20 +2,68 @@
...
 
+// Legacy CodeMirror 5 modes
+import { clojure } from "@codemirror/legacy-modes/mode/clojure";
+import { cmake } from "@codemirror/legacy-modes/mode/cmake";
+import { cobol } from "@codemirror/legacy-modes/mode/cobol";
+import { coffeeScript } from "@codemirror/legacy-modes/mode/coffeescript";
+import { commonLisp } from "@codemirror/legacy-modes/mode/commonlisp";
+import { d } from "@codemirror/legacy-modes/mode/d";
+// import { dart } from "@codemirror/legacy-modes/mode/dart";
+// import { django } from "@codemirror/legacy-modes/mode/django";
```
Comment (KonradStanski) [inline packages/web/src/hooks/useSyntaxHighlightingExtension.ts:36]: These exist on the https://codemirror.net/5/mode/ page but are seemingly missing from the import.

File: packages/web/src/hooks/useSyntaxHighlightingExtension.ts (hunk 1 above)
Comment (brendan-kellam) [reply packages/web/src/hooks/useSyntaxHighlightingExtension.ts:49]: That's strange - looks like it's here? https://github.com/codemirror/legacy-modes/blob/main/mode/octave.js

File: packages/web/src/hooks/useSyntaxHighlightingExtension.ts (hunk 2 above)
Comment (brendan-kellam) [reply packages/web/src/hooks/useSyntaxHighlightingExtension.ts:36]: Looks like dart is exported here: https://github.com/codemirror/legacy-modes/blob/main/mode/clike.js

I don't see django exported anywhere so not sure 🤔

File: packages/web/src/hooks/useSyntaxHighlightingExtension.ts [hunk 3]
```diff
@@ -30,28 +78,24 @@ export const useSyntaxHighlightingExtension = (language: string, view: EditorVie
 }
 
 export const getSyntaxHighlightingExtension = (language: string) => {
+    // maps linguist language defs to CodeMirror 6 language extensions
+    // and legacy CodeMirror 5 modes
+    // console.log(language.toLowerCase()); // uncomment to debug
```
Comment (brendan-kellam) [inline packages/web/src/hooks/useSyntaxHighlightingExtension.ts:83]: nit: we can probably just remove this commented out log, or change it to a `console.debug`

File: packages/web/src/hooks/useSyntaxHighlightingExtension.ts [hunk 4]
```diff
@@ -60,11 +104,109 @@ export const getSyntaxHighlightingExtension = (language: string) => {
...
+            return StreamLanguage.define(rpmSpec);
+        case "ruby":
+            return StreamLanguage.define(ruby);
+        case "scheme":
+            return StreamLanguage.define(scheme);
+        case "shell":
+            return StreamLanguage.define(shell);
+        case "swift":
+            return StreamLanguage.define(swift);
+        case "tcl":
```
Comment (brendan-kellam) [inline packages/web/src/hooks/useSyntaxHighlightingExtension.ts:198]: RE @NasaGeek @er0k : tcl syntax-highlighting support incoming thanks to @KonradStanski 😃

**Review by brendan-kellam — DISMISSED** [review]
LGTM thanks!

File: packages/web/src/hooks/useSyntaxHighlightingExtension.ts (hunk 3 above)
Comment (KonradStanski) [reply packages/web/src/hooks/useSyntaxHighlightingExtension.ts]: removed.

File: packages/web/src/hooks/useSyntaxHighlightingExtension.ts (hunk 2 above)
Comment (KonradStanski) [reply packages/web/src/hooks/useSyntaxHighlightingExtension.ts]: thanks for pointing out the clike ones. added.

File: packages/web/src/hooks/useSyntaxHighlightingExtension.ts (hunk 1 above)
Comment (KonradStanski) [reply packages/web/src/hooks/useSyntaxHighlightingExtension.ts]: Fixed and added

**Review by brendan-kellam — APPROVED** [review]

**Pull request merged**